packages/dataviews/src/components/dataviews/stories/index.story.tsx

```diff
@@ -105,3 +122,63 @@ export const FieldsNoSortableNoHidable = () => {
 		/>
 	);
 };
+
+/**
+ * Custom composition example
+ */
+function PlanetOverview( { planets }: { planets: SpaceObject[] } ) {
+	const moons = planets.reduce( ( sum, item ) => sum + item.satellites, 0 );
+
+	return (
+		<Card isBorderless style={ { padding: '12px 24px' } }>
+			<CardHeader>
+				<Heading level={ 2 }>{ __( 'Solar System numbers' ) }</Heading>
+			</CardHeader>
+
+			<CardBody>
+				<VStack spacing={ 2 }>
+					<Text size={ 18 } as="p">
+						{ createInterpolateElement(
+							_n(
+								'<PlanetsNumber /> planet',
+								'<PlanetsNumber /> planets',
+								planets.length
+							),
+							{
+								PlanetsNumber: (
+									<strong>{ planets.length } </strong>
+								),
+							}
+						) }
+					</Text>
+
+					<Text size={ 18 } as="p">
+						{ createInterpolateElement(
+							_n(
+								'<SatellitesNumber /> moon',
+								'<SatellitesNumber /> moons',
+								moons
+							),
+							{
+								SatellitesNumber: <strong>{ moons } </strong>,
+							}
+						) }
+					</Text>
+				</VStack>
+			</CardBody>
+		</Card>
+	);
+}
+
+export const FreeComposition = () => {
+	console.log( { data } );
+	const planets = data.filter( ( item ) =>
+		item.categories.includes( 'Planet' )
+	);
+
+	return (
+		<DataViews>
+			<PlanetOverview planets={ planets } />
```

Comment:
Here is where we'll add the DataViews subcomponents:

```ts
        <DataViews.Search />
        <DataViews.Filters />
        <DataViews.BulkActions />
        <DataViews.LayoutSwitcher />
        <DataViews.ViewConfig />
```

Reply:
Any children should access data via the context, not via props. This is important because data can be filtered, sorted, or paginated. Each component shouldn't need to reimplement this logic by itself. See how this [alternative approach](https://github.com/Automattic/wp-calypso/pull/102996#issuecomment-2841266284) handles data.

Reply:
That's partially correct from my PoV. We can rephrase it like:

> Any DataViews subcomponent children, such as Search, Filters, etc, should access data via the context, not via props. 

Because for custom children we have two options:

* expose the context, public from the API
* do not expose the context, and pass prop down too the children component

```ts
function Layout() {
  const data = useContext( DataViewsContext );
   return (
     <DataViews data={ data }>
       <CustomComponent data={ data } />
     </DataViews>
   );
 }
 ```

Reply:
The processed data (sorted, filtered, etc) is accessible through the public `filterSortAndPaginate()` helper